math_pendulum.cpp

```diff
@@ -2,12 +2,28 @@
 #include "vector.hpp"
 #include "json.hpp"
 #include "simulation_methods.hpp"
+#include <cmath>
 using json = nlohmann::json;
 
 
-double func(double t, Vector y, json conf){
+void func(double t, Vector* y, Vector* y_d, json* config){
```

Comment:
Why are you using pointers everywhere? Lets make sure you are well awar of how references could be used in cpp app here.

Also I'd suggest to make sure you do realize how to use const attributes properly.

Reply:
This function is the given equation and is called like several millions times, so I think it is faster to create pointers to objects rather than copies of these objects. Maybe there is a way to pass nothing at all to this function but I cant think of a way.

Reply:
I meant you are using c-style pointers instead of cpp-style references&. I miself don't mind using c style for any reason. Just keep in mind that such code-style decisions should be consistent so avoid mixing *a and &b in function interfaces for no reason (there are cases when such constructions are a must).

Reply:
Oh, sorry I forgot they exist. Yeah I think it is better to use references.